src/ReaderScreen/components/AudioPlayer/utils/getSequenceFromPosition.js

```diff
@@ -0,0 +1,60 @@
+import { logError } from "@common";
+import fetchLRCData from "./fetchLRC";
+
+/**
+ * Get the current sequence number from lyrics JSON based on audio position
+ * @param {string} lyricsUrl - URL to the lyrics JSON file
+ * @param {number} position - Current audio position in seconds
+ * @returns {Promise<number|null>} - Sequence number or null if not found
+ */
+const getSequenceFromPosition = async (lyricsUrl, position) => {
+  try {
+    if (!lyricsUrl || position == null) {
+      return null;
+    }
+
+    const lrcData = await fetchLRCData(lyricsUrl);
+    if (!lrcData || !Array.isArray(lrcData) || lrcData.length === 0) {
+      return null;
+    }
+
+    // Find the timestamp entry that contains the current position
+    const currentTimestamp = lrcData.find(
+      (timestamp) => position >= timestamp.start && position <= timestamp.end
+    );
+
+    return currentTimestamp ? currentTimestamp.sequence : null;
+  } catch (error) {
+    logError("Error getting sequence from position:", error);
+    return null;
+  }
+};
+
+/**
+ * Get the position (start time) for a given sequence number from lyrics JSON
+ * @param {string} lyricsUrl - URL to the lyrics JSON file
+ * @param {number} sequence - Sequence number to find
+ * @returns {Promise<number|null>} - Start position in seconds or null if not found
+ */
+const getPositionFromSequence = async (lyricsUrl, sequence) => {
+  try {
+    if (!lyricsUrl || sequence == null) {
+      return null;
+    }
+
+    const lrcData = await fetchLRCData(lyricsUrl);
+    if (!lrcData || !Array.isArray(lrcData) || lrcData.length === 0) {
+      return null;
+    }
```

Comment:
This code is being repeated in both functions, we can create another function `loadLrc` for this.